array-manipulation-3/src/zip.js

```diff
@@ -1 +1,15 @@
 /* eslint-disable no-unused-vars */
+
+function zip(first, second) {
+  var mainArray = [];
+  for (var i = 0; i < first.length; i++) {
+    var tempArray = [];
+    if (first[i] !== undefined && second[i] !== undefined) {
```

Comment:
Instead of checking for `undefined` (since its a legit value), change up your `for` loop to stop at the shorter of the two

```js
for (var i = 0; i < first.length && i < second.length; i++) {
```